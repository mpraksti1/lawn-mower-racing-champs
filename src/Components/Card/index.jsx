import React from 'react';
import PropTypes from 'prop-types';
import Text from '../../Elements/Text';
import StyledDiv from './styles';

const Card = (props) => {
  const {
    children,
    title,
    copy,
  } = props;

  return (
    <StyledDiv {...props}>
      <div className="icon">{children}</div>
      <Text className="title" sans block alignCenter xlg>{title}</Text>
      <Text className="copy" sans block alignCenter thin spaceAround>{copy}</Text>
    </StyledDiv>
  );
};

Card.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  copy: PropTypes.string.isRequired,
  bg: PropTypes.string.isRequired,
};

export default Card;
