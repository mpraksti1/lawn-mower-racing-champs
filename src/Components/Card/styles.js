import styled from 'styled-components';

const StyledDiv = styled.div`
  background: url(${({ bg }) => bg});
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;

  .icon {
    text-align: center;
    
    svg {
      fill: ${({ theme }) => theme.colors.green};
      height: 72px;
      width: 72px;
    }
  }
`;

export default StyledDiv;
