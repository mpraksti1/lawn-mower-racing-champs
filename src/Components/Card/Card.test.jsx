import CommunityIcon from 'react-icons/lib/md/people';
import Card from '.';

describe('Card component', () => {
  it('renders without crashing', () => {
    shallow(
      <Card title="I am a card!" copy="A really good card!" bg="http://www.abc.net.au/news/image/7202178-3x2-940x627.jpg">
        <CommunityIcon />
      </Card>,
    );
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <Card title="I am a card!" copy="A really good card!" bg="http://www.abc.net.au/news/image/7202178-3x2-940x627.jpg">
        <CommunityIcon />
      </Card>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
