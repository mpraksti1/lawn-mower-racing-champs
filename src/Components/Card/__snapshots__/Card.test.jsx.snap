// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches snapshot 1`] = `
<styled.div
  bg="http://www.abc.net.au/news/image/7202178-3x2-940x627.jpg"
  copy="A really good card!"
  title="I am a card!"
>
  <div
    className="icon"
  >
    <MdPeople />
  </div>
  <Text
    alignCenter={true}
    block={true}
    className="title"
    sans={true}
    xlg={true}
  >
    I am a card!
  </Text>
  <Text
    alignCenter={true}
    block={true}
    className="copy"
    sans={true}
    spaceAround={true}
    thin={true}
  >
    A really good card!
  </Text>
</styled.div>
`;
